pkg/postrender/postrender.go

```diff
@@ -26,4 +26,12 @@ type PostRenderer interface {
 	// expects the modified results to be returned on a separate buffer or an
 	// error if there was an issue or failure while running the post render step
 	Run(renderedManifests *bytes.Buffer) (modifiedManifests *bytes.Buffer, err error)
+	// RunIncHooks expects and returns a map of file names and their rendered contents
+	// Example:
+	// > map[string]string{
+	// >   "templates/foo.yaml": "Kind: Pod..",
+	// >   "templates/baz.yaml": "Kind: ConfigMap...",
+	// > }
+	RunIncHooks(renderedFiles map[string]string) (modifiedFiles map[string]string, err error)
+	IncHooks() bool
```

Comment:
This will break any existing 3rd party software that implements this interface, which might be ok since we're on the helm 4 path.

Reply:
Would [this change](https://github.com/helm/helm/pull/13154/commits/7f3e88d038dbd5d489a907649880793e8b54887a) be appropriate to keep the interface backwards compatible?

I was going to rebase the MR but noticed that main is moving to helm v4. Should I rebase on top of dev-v3 instead?

Reply:
Hi @joejulian 
Can I get a review on this MR? Anything further I can do to help move it along?

Reply:
> Would [this change](https://github.com/helm/helm/pull/13154/commits/7f3e88d038dbd5d489a907649880793e8b54887a) be appropriate to keep the interface backwards compatible?
> 
> I was going to rebase the MR but noticed that main is moving to helm v4. Should I rebase on top of dev-v3 instead?

No, this is fine. I wanted to confirm with some of the other maintainers.

Reply:
@joejulian Let me know what the next steps are to move this forward.